From b61cbed6895fcd3eae4c8458a69995870a22a5e0 Mon Sep 17 00:00:00 2001
From: Vincent Blut <vincent.debian@free.fr>
Date: Wed, 12 Jan 2022 18:08:34 +0100
Subject: test: ensure awk commands in 008-ntpera return an integer

Some awk interpreters (e.g. mawk) print long integers in exponential
notation skewing the test result.

Bug: https://listengine.tuxfamily.org/chrony.tuxfamily.org/chrony-dev/2022/01/msg00001.html
Applied-Upstream: https://git.tuxfamily.org/chrony/chrony.git/commit/?id=b61cbed6895fcd3eae4c8458a69995870a22a5e0
Last-Update: 2022-01-13
diff --git a/test/simulation/008-ntpera b/test/simulation/008-ntpera
index 2a4f332..2eea63b 100755
--- a/test/simulation/008-ntpera
+++ b/test/simulation/008-ntpera
@@ -29,7 +29,7 @@ echo "$ntp_start" | grep -q '-' && test_skip
 
 for time_offset in -1e-1 1e-1; do
 	for start_offset in 0 "2^32 - $limit"; do
-		export CLKNETSIM_START_DATE=$(awk "BEGIN {print $ntp_start + $start_offset}")
+		export CLKNETSIM_START_DATE=$(awk "BEGIN {printf \"%.0f\", $ntp_start + $start_offset}")
 		run_test || test_fail
 		check_chronyd_exit || test_fail
 		check_source_selection || test_fail
@@ -38,7 +38,7 @@ for time_offset in -1e-1 1e-1; do
 	done
 
 	for start_offset in -$limit "2^32"; do
-		export CLKNETSIM_START_DATE=$(awk "BEGIN {print $ntp_start + $start_offset}")
+		export CLKNETSIM_START_DATE=$(awk "BEGIN {printf \"%.0f\", $ntp_start + $start_offset}")
 		run_test || test_fail
 		check_chronyd_exit || test_fail
 		check_source_selection || test_fail
-- 
cgit v0.10.2

